NameLayerMain/src/vg/civcraft/mc/namelayer/group/Group.java

```diff
@@ -490,21 +513,37 @@ public void changeDefaultGroup(UUID uuid) {
 	 * Sets the password for a group. Set the parameter as null to remove the password.
 	 * @param password- The password of the group.
 	 */
-	public void setPassword(String password) {
+	public void setPassword(String password){
+		setPassword(password,true);
+	}
+	public void setPassword(String password, boolean savetodb) {
 		this.password = password;
-		db.updatePassword(name, password);
+		if (savetodb){
+			db.updatePasswordAsync(name, password);
+			Mercury.message("setPass " + this.name + " " + password);
+		}
 	}
 
 	/**
 	 * Sets the owner of the group.
 	 * @param uuid- The UUID of the Player.
 	 */
-	public void setOwner(UUID uuid) {
+	public void setOwner(UUID uuid){
+		setOwner(uuid,true);
+	}
+	
+	public void setOwner(UUID uuid, boolean savetodb) {
 		this.owner = uuid;
-		db.setFounder(uuid, this);
+		if (savetodb){
+			db.setFounderAsync(uuid, this);
+			Mercury.message("setOwner "+ this.name + " " + uuid.toString());
+		}
 	}
 
-	public void setDisciplined(boolean value) { this.isDisciplined = value; }
+	public void setDisciplined(boolean value) {
+		this.isDisciplined = value;
+		Mercury.message("discipline "+ this.name +" "+ this.isDisciplined);
```

Comment:
Worth noting that the MercuryListener doesn't have a handler for messages of type `discipline`.